collector/perf_schema_replication_group_member_stats.go

```diff
@@ -16,42 +16,51 @@ package collector
 import (
 	"context"
 	"database/sql"
+	"strconv"
 
 	"github.com/go-kit/kit/log"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
-const perfReplicationGroupMemeberStatsQuery = `
-	SELECT MEMBER_ID,COUNT_TRANSACTIONS_IN_QUEUE,COUNT_TRANSACTIONS_CHECKED,COUNT_CONFLICTS_DETECTED,COUNT_TRANSACTIONS_ROWS_VALIDATING
-	  FROM performance_schema.replication_group_member_stats
-	`
+const perfReplicationGroupMemberStatsQuery = `
+	SELECT * FROM performance_schema.replication_group_member_stats WHERE MEMBER_ID=@@server_uuid
+`
 
-// Metric descriptors.
 var (
-	performanceSchemaReplicationGroupMemberStatsTransInQueueDesc = prometheus.NewDesc(
-		prometheus.BuildFQName(namespace, performanceSchema, "transaction_in_queue"),
-		"The number of transactions in the queue pending conflict detection checks. Once the "+
-			"transactions have been checked for conflicts, if they pass the check, they are queued to be applied as well.",
-		[]string{"member_id"}, nil,
-	)
-	performanceSchemaReplicationGroupMemberStatsTransCheckedDesc = prometheus.NewDesc(
-		prometheus.BuildFQName(namespace, performanceSchema, "transaction_checked"),
-		"The number of transactions that have been checked for conflicts.",
-		[]string{"member_id"}, nil,
-	)
-	performanceSchemaReplicationGroupMemberStatsConflictsDetectedDesc = prometheus.NewDesc(
-		prometheus.BuildFQName(namespace, performanceSchema, "conflicts_detected"),
-		"The number of transactions that did not pass the conflict detection check.",
-		[]string{"member_id"}, nil,
-	)
-	performanceSchemaReplicationGroupMemberStatsTransRowValidatingDesc = prometheus.NewDesc(
-		prometheus.BuildFQName(namespace, performanceSchema, "transaction_rows_validating"),
-		"The current size of the conflict detection database (against which each transaction is certified).",
-		[]string{"member_id"}, nil,
-	)
+	// The list of columns we are interesting in.
+	// In MySQL 5.7 these are the 4 first columns available. In MySQL 8.x all 8.
+	perfReplicationGroupMemberStats = map[string]struct {
+		vtype prometheus.ValueType
+		desc  *prometheus.Desc
+	}{
+		"COUNT_TRANSACTIONS_IN_QUEUE": {prometheus.GaugeValue,
+			prometheus.NewDesc(prometheus.BuildFQName(namespace, performanceSchema, "transactions_in_queue"),
+				"The number of transactions in the queue pending conflict detection checks.", nil, nil)},
+		"COUNT_TRANSACTIONS_CHECKED": {prometheus.CounterValue,
+			prometheus.NewDesc(prometheus.BuildFQName(namespace, performanceSchema, "transactions_checked"),
```

Comment:
Sorry, another naming nitpick. :)

`CounterValue` metrics should be named `_total`.